api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -119,6 +119,16 @@ public interface RewriteDataFiles
 
   String REWRITE_JOB_ORDER_DEFAULT = RewriteJobOrder.NONE.orderName();
 
+  /**
+   * Constant representing the target output partition specification ID.
+   *
+   * <p>output-spec-id ID is used by the file rewriter during the rewrite operation to identify the
+   * specific output partition spec. This allows rewriting files into a partitioning which is not
```

Comment:
instead of "this allows ...." 

Data will be reorganized during the rewrite to align with the output partitioning. Defaults to the current table specification.